### DISCUSSION TITLE
Filter files help me

### DISCUSSION THREAD

**Discussion by Luiio01** [aahnik/tgcf#305] [General]
Hi I followed the tutorial if I want to put only files by doing so

admins: [my id]
forwards: 
           - source: id Channel
             dest: [id Channel]

plugins:
     filter:
          files:
       whitelist: [document]

But with any voice even photo or gif doesn't work where I'm wrong thanks

**Comment by tissole** [comment]
If you're trying to forward only files and the gifs and voice msg are forwarded too, you can't do anything now. I encounter this behaviour when I first tested the filtering feature, see #131. Maybe in the future filtering by file extension will resolve this issue.

**Reply by Luiio01** [reply]
if i want to block all media like gif sticker video and i just want the files?

**Reply by Luiio01** [reply]
If i write

plugin:
     files:
       whitelist: for file?

**Reply by tissole** [reply]
You can block these categories: audio, gif, video, video_note, sticker, contact, photo, document, nofile.
Nofile means text msg, conversations. But gifs that exist as files, not as photos can't be filtered. These are considered documents and will be forwarded. Same with voice msgs.
